starter-code/src/App.js

```diff
@@ -1,8 +1,37 @@
 import React, { Component } from 'react';
 import logo from './logo.svg';
 import './App.css';
+import contacts from './contacts.json';
+
+
+class Contacts extends Component {
+  constructor(props) {
+    super(props);
+  }
+  render() {
+    return (
+      <tr>
+        <td><img src={this.props.picture} /></td>
+        <td>{this.props.name}</td>
+        <td>{this.props.popularity}</td>
+      </tr>
+    )
+
+  }
+}
+
+
 
 class App extends Component {
+  constructor(props) {
+    super(props);
+    this.state = { contacts, compteur:5 }
+    this.handleClick = this.handleClick.bind(this)
+  }
+
+  handleClick() {
+    this.setState({compteur: this.state.compteur+1 })
+  }
```

Comment:
Well done for the adding contact, although technically your added contact is not random 😉 . Here's an example of how you could have handled this:
```js
  AddRandomActor() {
    const randomNum = Math.floor(Math.random() * contacts.length)
    const newContact = contacts[randomNum]
    const updatedContactList = this.state.contacts.slice()
    updatedContactList.push(newContact)
    this.setState({ contacts: updatedContactList })
  }
```